react/features/chat/components/native/ChatButton.js

```diff
@@ -0,0 +1,130 @@
+// @flow
+
+import { connect } from 'react-redux';
+
+import { getLocalParticipant } from '../../../base/participants';
+import {
+    AbstractButton,
+    type AbstractButtonProps
+} from '../../../base/toolbox';
+import { openDisplayNamePrompt } from '../../../display-name';
+
+import { toggleChat } from '../../actions';
+import { getUnreadCount } from '../../functions';
+
+type Props = AbstractButtonProps & {
+
+    /**
+     * Function to display chat.
+     *
+     * @protected
+     */
+    _displayChat: Function,
+
+    /**
+     * Function to diaply the name prompt before displaying the chat
+     * window, if the user has no display name set.
+     */
+    _displayNameInputDialog: Function,
+
+    /**
+     * Whether or not to block chat access with a nickname input form.
+     */
+    _showNamePrompt: boolean,
+
+    /**
+     * The unread message count.
+     */
+    _unreadMessageCount: number
+};
+
+/**
+ * Implements an {@link AbstractButton} to open the chat screen on mobile.
+ */
+class ChatButton extends AbstractButton<Props, *> {
+    accessibilityLabel = 'toolbar.accessibilityLabel.chat';
+    iconName = 'chat';
+    label = '';
+    toggledIconName = 'chat-unread';
+
+    /**
+     * Handles clicking / pressing the button, and opens the appropriate dialog.
+     *
+     * @private
+     * @returns {void}
+     */
+    _handleClick() {
+        if (this.props._showNamePrompt) {
+            this.props._displayNameInputDialog(() => {
+                this.props._displayChat();
+            });
+        } else {
+            this.props._displayChat();
+        }
+    }
+
+    /**
+     * Renders the button toggled when there are unread messages.
+     *
+     * @protected
+     * @returns {boolean}
+     */
+    _isToggled() {
+        return Boolean(this.props._unreadMessageCount);
+    }
+}
+
+/**
+ * Maps redux actions to {@link InviteButton}'s React
+ * {@code Component} props.
+ *
+ * @param {Function} dispatch - The redux action {@code dispatch} function.
+ * @returns {{
+ *     _onAddPeople,
+ *     _onShareRoom
```

Comment:
I wonder where you copied this from.

Reply:
From nowhere, of course, and the simple fact that you thought that I copied this is outrageous 😱